extensions/src/main/java/org/pentaho/platform/web/http/api/resources/ActionResource.java

```diff
@@ -145,6 +149,8 @@ IAction createActionBean( final String actionClass, final String actionId ) thro
 
     public void run() {
       try {
+        Optional.ofNullable( mdcContextMap ).ifPresent( s -> MDC.setContextMap( mdcContextMap ) );
```

Comment:
somewhat more self-explanatory lambda, that makes it less cryptic?... for your consideration:

        Optional.ofNullable( mdcContextMap ).ifPresent( map -> MDC.setContextMap( map ) );